---
title: C26111
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C26111
helpviewer_keywords:
- C26111
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 5541ec5dcc8ccd55fe5288265090e3d9c5958e04
ms.sourcegitcommit: 21d667104199c2493accec20c2388cf674b195c3
ms.translationtype: MT
ms.contentlocale: de-DE
ms.lasthandoff: 02/08/2019
ms.locfileid: "55947912"
---
# <a name="c26111"></a>C26111
Warnung C26111: Fehler des Aufrufers beim Sperre \<Sperre > vor dem Aufruf von Funktion \<Func >.

 Die Anmerkung `_Requires_lock_not_held_` erzwingt eine Voraussetzung, dass die Anzahl der Sperren für die angegebene Sperre größer als 0 (null) sein kann, wenn die Funktion aufgerufen wird. Warnung C26111 wird ausgegeben, wenn eine Funktion fehlschlägt, um die Sperre freizugeben, bevor sie eine andere Funktion aufruft.

## <a name="example"></a>Beispiel
 Im folgende Beispiel wird die Warnung C26111 generiert, da die `_Requires_lock_not_held_` Vorbedingung verletzt wird, durch den Aufruf von `DoNotLock` im gesperrten Abschnitt.

```

typedef struct _DATA
{
    CRITICAL_SECTION cs;
    int d;
} DATA;

_Requires_lock_not_held_(p->cs)

void DoNotLock(DATA* p)
{
    EnterCriticalSection(&p->cs);
    p->d = 0;
    LeaveCriticalSection(&p->cs);
}

void LockedFunction(DATA* p)
{
    EnterCriticalSection(&p->cs);
    DoNotLock(p); // Warning C26111
    LeaveCriticalSection(&p->cs);
}
```